### PR TITLE
add support for generics in CRD generation

### PR DESCRIPTION
**Pull request by euberseder-hubspot**
Adds support for traversal of generic classes.

### CODE REVIEW

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java [hunk 1]
```diff
@@ -363,6 +363,47 @@ private Map<String, Method> indexPotentialAccessors(TypeDef definition) {
...
+        }
+      }
+      return original;
+    }
+
+    static ParameterMap from(ClassRef classRef, ParameterMap parentMappings) {
+      TypeDef def = Types.typeDefFrom(classRef);
+
+      Map<String, TypeRef> mappings = new HashMap<>();
+      if(!def.getParameters().isEmpty()) {
```
Comment (bequinn-hubspot) [inline crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java]: This check seems unnecessary

File: crd-generator/api/src/test/java/io/fabric8/crd/example/generic/NestedGeneric.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package io.fabric8.crd.example.generic;
+
+import java.util.List;
+
+public class NestedGeneric<P> {
+  Generic<P> quux;
+  List<P> corge;
```
Comment (bequinn-hubspot) [inline crd-generator/api/src/test/java/io/fabric8/crd/example/generic/NestedGeneric.java:7]: never seen corge before :)

**Review by bequinn-hubspot — APPROVED** [review]
This makes sense to me 👍

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java (hunk 1 above)
Comment (euberseder-hubspot) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java]: Indeed.

**Pull request merged**